### ISSUE TITLE
Integrate foundation models for automatic novelty detection and pattern suggestions

### ISSUE THREAD

**Issue by rjwalters** [rulehunt/rulehunt#179]
## Problem

Human ratings provide high-quality signals but don't scale to explore 2^140 rules. ADANA proposes training ALife-specific foundation models on human preference data to automate interesting-ness detection at scale.

## Proposal

Integrate vision-language models (VLMs) or vision transformers (ViTs) to automatically score patterns and guide exploration toward novel regions of rule space.

## Acceptance Criteria

- [ ] API endpoint to submit patterns to ADANA dataset (when available)
- [ ] Integration with CLIP/ViT embeddings for novelty scoring
- [ ] "Suggest interesting regions" feature using ML-guided exploration
- [ ] Automated interestingness scoring as alternative/supplement to human ratings
- [ ] Embeddings-based similarity search (find patterns similar to current one)
- [ ] Novelty metric: measure distance in embedding space from previously seen patterns

## Implementation Guidance

**Phase 1: ADANA Dataset Contribution**
```typescript
// Submit pattern to ADANA archive
async function submitToADANA(pattern: Pattern, rating: number) {
  await fetch('https://adana-api.example.com/submit', {
    method: 'POST',
    body: JSON.stringify({
      system: 'rulehunt',
      rule: pattern.rule,
      snapshot: pattern.gridState,
      humanRating: rating,
      metadata: pattern.statistics
    })
  })
}
```

**Phase 2: ML-Guided Exploration**
```typescript
// Use CLIP embeddings to score novelty
async function scoreNovelty(gridImage: ImageData): Promise<number> {
  const embedding = await getCLIPEmbedding(gridImage)
  const nearestNeighbors = findKNN(embedding, seenPatterns, k=10)
  const noveltyScore = avgDistance(embedding, nearestNeighbors)
  return noveltyScore
}
```

**Phase 3: Intelligent Search**
- Weight random rule sampling by predicted interestingness
- Prioritize rule regions with high novelty scores
- Implement Quality-Diversity (QD) algorithm using ML features

**Required Infrastructure:**
- Model serving (CLIP API or local inference)
- Embedding database (vector similarity search)
- Exploration state tracking (which rules sampled, what found)

**Why This Matters:**
- Scales human judgment to vast rule spaces
- Aligns with ADANA's goal of "interest-aware scoring function"
- Enables true open-ended exploration guided by learned priors
- Creates feedback loop: human ratings → train model → better exploration

## Related

- Parent: #175 (ADANA document addition)
- Depends on: #176 (human rating data), #177 (export infrastructure)
- See [ADANA.md](docs/ADANA.md) sections on "Fine-tune and adapt foundation models" and "Foundation Model" considerations

## Priority

Low (Research) - Requires ADANA dataset infrastructure and trained models

## Open Questions

1. Should we use pre-trained CLIP or wait for ADANA-specific fine-tuned model?
2. How often to query ML model vs relying on cached scores?
3. Client-side inference (WASM) vs server-side API calls?
4. Can we contribute to ADANA model training with our pattern/rating data?
